### ISSUE TITLE
Run mpMetrics-2.0 TCK against Open Liberty

### ISSUE THREAD

**Issue by fmhwong** [OpenLiberty/open-liberty#6189]
Get OpenLiberty to be able to run entire TCK cleanly using mpMetrics-2.0 beta feature (including ConcurrentGauge tests)

**Comment by fmhwong** [comment]
On Java 11 build, the following tests failed.

```
Tests run: 47, Failures: 0, Errors: 6, Skipped: 0, Time elapsed: 12.456 sec <<< FAILURE! - in org.eclipse.microprofile.metrics.test.MpMetricTest
testBase(org.eclipse.microprofile.metrics.test.MpMetricTest)  Time elapsed: 0.188 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.MpMetricTest.testBase(MpMetricTest.java:211)

testBaseAttributeJson(org.eclipse.microprofile.metrics.test.MpMetricTest)  Time elapsed: 0.077 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.MpMetricTest.testBaseAttributeJson(MpMetricTest.java:243)

testBaseSingularMetricsPresent(org.eclipse.microprofile.metrics.test.MpMetricTest)  Time elapsed: 0.081 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.MpMetricTest.testBaseSingularMetricsPresent(MpMetricTest.java:260)

testApplicationMetricsJSON(org.eclipse.microprofile.metrics.test.MpMetricTest)  Time elapsed: 0.125 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.MpMetricTest.testApplicationMetricsJSON(MpMetricTest.java:506)

testMultipleTaggedMetricsJSON(org.eclipse.microprofile.metrics.test.MpMetricTest)  Time elapsed: 0.116 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.MpMetricTest.testMultipleTaggedMetricsJSON(MpMetricTest.java:1058)

testTranslateSemiColonToUnderScoreJSON(org.eclipse.microprofile.metrics.test.MpMetricTest)  Time elapsed: 0.124 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.MpMetricTest.testTranslateSemiColonToUnderScoreJSON(MpMetricTest.java:1206)

Running org.eclipse.microprofile.metrics.test.ReusableMetricsTest
Tests run: 7, Failures: 0, Errors: 3, Skipped: 0, Time elapsed: 3.342 sec <<< FAILURE! - in org.eclipse.microprofile.metrics.test.ReusableMetricsTest
testSharedCounter(org.eclipse.microprofile.metrics.test.ReusableMetricsTest)  Time elapsed: 0.053 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.ReusableMetricsTest.testSharedCounter(ReusableMetricsTest.java:123)

testSharedCounterAgain(org.eclipse.microprofile.metrics.test.ReusableMetricsTest)  Time elapsed: 0.037 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.ReusableMetricsTest.testSharedCounterAgain(ReusableMetricsTest.java:152)

testReusedHistogram(org.eclipse.microprofile.metrics.test.ReusableMetricsTest)  Time elapsed: 0.042 sec  <<< ERROR!
java.lang.ClassCastException: [B incompatible with [C
	at groovy.json.internal.FastStringUtils$StringImplementation$1.toCharArray(FastStringUtils.java:88)
	at groovy.json.internal.FastStringUtils.toCharArray(FastStringUtils.java:175)
	at groovy.json.internal.CharBuf.addJsonFieldName(CharBuf.java:543)
	at groovy.json.JsonOutput.writeMap(JsonOutput.java:424)
	at groovy.json.JsonOutput.toJson(JsonOutput.java:204)
	at com.jayway.restassured.path.json.JsonPath.toJsonString(JsonPath.java:979)
	at com.jayway.restassured.path.json.JsonPath.prettify(JsonPath.java:613)
	at org.eclipse.microprofile.metrics.test.ReusableMetricsTest.testReusedHistogram(ReusableMetricsTest.java:181)
```

**Comment by fmhwong** [comment]
According to https://issues.apache.org/jira/browse/GROOVY-7716, this should be fixed in version 2.4.6 but Microprofile Metrics is still using 2.4.0.

**Comment by fmhwong** [comment]
Opened `https://github.com/eclipse/microprofile-metrics/issues/391` and PR `https://github.com/eclipse/microprofile-metrics/pull/392`
